app/Observers/CampaignObserver.php

```diff
@@ -170,23 +173,30 @@ public function deleted(Campaign $campaign)
      */
     public function deleting(Campaign $campaign)
     {
-        // Technically, only a campaign with a single user can be deleted.
-        foreach ($campaign->members as $member) {
-            $member->user->notify(new Header(
-                'campaign.deleted',
-                'trash',
-                'yellow',
-                [
-                    'campaign' => $campaign->name
-                ]
-            ));
-            $member->delete();
-        }
-
-        // Delete the campaign setting
-        $campaign->setting->delete();
 
-        ImageService::cleanup($campaign, 'header_image');
+        if ($campaign->isForceDeleting()) {
+            // Technically, only a campaign with a single user can be deleted.
+            foreach ($campaign->members as $member) {
+                $member->delete();
+            }
+            // Delete the campaign settings.
+            $campaign->setting->delete();
+
+            ImageService::cleanup($campaign, 'header_image');
```

Comment:
If using the observer, check if we need ImageService::cleanup() called twice for the header image and the campaign image. Or maybe move all of that to the Job to keep it cleaner?